src/components/AudioPlayer/AudioPlayer.js

```diff
@@ -0,0 +1,205 @@
+import { useEffect, useRef, useState } from 'react'
+
+import AudioControls from '../AudioControls/AudioControls.js'
+import PropTypes from 'prop-types'
+import styled from 'styled-components/macro'
+
+AudioPlayer.propTyped = {
+  audioObject: PropTypes.object,
+}
+
+export default function AudioPlayer({ audioObject }) {
+  const [progress, setProgress] = useState(0)
+  const [isPlaying, setIsPlaying] = useState(false)
+  const [isReady, setIsReady] = useState(false)
+
+  const { image, title, author, audioSrc } = audioObject
+  const audioRef = useRef(new Audio(audioSrc))
+  const intervalRef = useRef()
+
+  const { duration } = audioRef.current
+
+  // some setup before playing
+  useEffect(() => {
+    audioRef.current = new Audio(audioSrc)
+    setProgress(audioRef.current.currentTime)
+
+    if (isReady) {
+      audioRef.current.play()
+      setIsPlaying(true)
+      startTimer()
+    }
+    if (!isReady) setIsReady(true)
+  }, [isReady, audioSrc])
```

Comment:
maybe you can put this in a hook with a speaking name, then you don't need those comments...

Reply:
Yes that is true. I will think about that. My thought process is, that it is easier to understand when everything is like that. Because i dont need those values anywhere else but this file.